client/landing/stepper/declarative-flow/internals/steps-repository/completing-purchase/index.tsx

```diff
@@ -39,60 +40,84 @@ const CompletingPurchase: Step = function CompletingPurchase( { navigation, flow
 		}
 
 		// convert ArrayBuffer to Blob
-		const blob = new Blob( [ buffer ], { type: type } );
-
-		return blob;
+		return new Blob( [ buffer ], { type: type } );
 	};
 
-	const completeLinkInBioFlow = () => {
-		if ( site ) {
-			setPendingAction( async () => {
-				setProgress( 0 );
-				setProgressTitle( __( 'Completing Purchase' ) );
-				const state = getState();
-				await saveSiteSettings( site.ID, {
-					blogname: state.siteTitle,
-					blogdescription: state.siteDescription,
-				} );
-
-				recordTracksEvent( 'calypso_signup_site_options_submit', {
-					has_site_title: !! state.siteTitle,
-					has_tagline: !! state.siteDescription,
-				} );
-				setProgress( 0.5 );
-				setProgressTitle( __( 'Creating your Link in Bio' ) );
-
-				await wait( 1000 );
-
-				setProgress( 0.8 );
-				setProgressTitle( __( 'Preparing Next Steps' ) );
-
-				if ( state.siteLogo ) {
-					try {
-						await setSiteLogo(
-							new File( [ base64ImageToBlob( state.siteLogo ) ], 'site-logo.png' )
-						);
-					} catch ( _error ) {
-						// communicate the error to the user
-					}
-				}
-				setProgress( 1 );
-			} );
+	const postSiteSettings = useCallback( async () => {
+		if ( ! site ) {
+			return;
+		}
+
+		await saveSiteSettings( site.ID, {
+			blogname: state.siteTitle,
+			blogdescription: state.siteDescription,
+		} );
+
+		recordTracksEvent( 'calypso_signup_site_options_submit', {
+			has_site_title: !! state.siteTitle,
+			has_tagline: !! state.siteDescription,
+		} );
+	}, [ site, saveSiteSettings, state ] );
+
+	const postSiteLogo = useCallback( async () => {
+		if ( ! state.siteLogo ) {
+			return;
+		}
+
+		try {
+			await setSiteLogo( new File( [ base64ImageToBlob( state.siteLogo ) ], 'site-logo.png' ) );
+		} catch ( _error ) {
+			// communicate the error to the user
 		}
+	}, [ setSiteLogo, state ] );
+
+	const completeLinkInBioFlow = () => {
+		setPendingAction( async () => {
+			setProgress( 0 );
+			setProgressTitle( __( 'Completing Purchase' ) );
+			await postSiteSettings();
+
+			setProgress( 0.5 );
+			setProgressTitle( __( 'Creating your Link in Bio' ) );
+
+			await wait( 1000 );
+
+			setProgress( 0.8 );
+			setProgressTitle( __( 'Preparing Next Steps' ) );
+
+			await postSiteLogo();
+			setProgress( 1 );
+			return 'launchpad';
+		} );
+	};
+
+	const completeNewsletterFlow = () => {
+		setPendingAction( async () => {
```

Comment:
Saved in site settings ✌️

Reply:
@escapemanuele saving of the accent color overlaps with the @Automattic/apex 's project: pdDOJh-Ee-p2.

Our initial thought was to store it directly as a theme's primary color. I am not sure how this can be done just yet but it seems to be doable since it can be done by editing global styles in FSE. I am looking into this right now.